sharding/client/smc.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/sharding"
+	"github.com/ethereum/go-ethereum/sharding/contracts"
+)
+
+// initSMC initializes the sharding manager contract bindings.
+// If the SMC does not exist, it will be deployed.
+func initSMC(c *shardingClient) (*contracts.SMC, error) {
```

Comment:
Can we put this function in the client.go file? The client is the one that inits the SMC and it doesn't make sense for this file to exist at all if we only envision having a few functions or just 1 in it.

Reply:
Sure. I don't feel very strongly about having this method in its own file so I'll move it.

Reply:
Done